wasm/jit/jit_amd64.go

```diff
@@ -1096,6 +1098,90 @@ func (b *amd64Builder) handleLe(o *wazeroir.OperationLe) error {
 	return nil
 }
 
+func (b *amd64Builder) handleLoad(o *wazeroir.OperationLoad) error {
+	base := b.locationStack.pop()
+	if base.onStack() {
+		if err := b.moveStackToRegisterWithAllocation(generalPurposeRegisterTypeInt, base); err != nil {
+			return err
+		}
+	} else if base.onConditionalRegister() {
+		if err := b.moveConditionalToGPRegister(base); err != nil {
+			return err
+		}
+	}
+
+	// At this point, base's value is on the integer general purpose reg.
+	// We reuse the register below, so we alias it here for readability.
+	reg := base.register
+
+	// Then we have to calculate the offset on the memory region.
+	addOffsetToBase := b.newProg()
+	addOffsetToBase.As = x86.AADDL // 32-bit!
+	addOffsetToBase.To.Type = obj.TYPE_REG
+	addOffsetToBase.To.Reg = reg
+	addOffsetToBase.From.Type = obj.TYPE_CONST
+	addOffsetToBase.From.Offset = int64(o.Arg.Offest)
+	b.addInstruction(addOffsetToBase)
+
+	// TODO: Emit instructions here to check memory out of bounds as
+	// potentially it would be an security risk.
+
+	var (
+		isIntType bool
+		movInst   obj.As
+	)
+	switch o.Type {
+	case wazeroir.SignLessTypeI32:
+		isIntType = true
+		movInst = x86.AMOVL
+	case wazeroir.SignLessTypeI64:
+		isIntType = true
+		movInst = x86.AMOVQ
+	case wazeroir.SignLessTypeF32:
+		isIntType = false
+		movInst = x86.AMOVL
+	case wazeroir.SignLessTypeF64:
+		isIntType = false
+		movInst = x86.AMOVQ
+	}
+
+	if isIntType {
+		// For integer types, we just read the corresponding bytes from the offset to the memory
```

Comment:
"just" makes things seem simple and even if it is, let's avoid it. I saw a presentation once where the speaker called "just" a magician!

```suggestion
		// For integer types, read the corresponding bytes from the offset to the memory
```